universe/world/calendar/calendar.go

```diff
@@ -0,0 +1,195 @@
+package calendar
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/momentum-xyz/posbus-protocol/posbus"
+	"github.com/pkg/errors"
+
+	"github.com/momentum-xyz/ubercontroller/logger"
+	"github.com/momentum-xyz/ubercontroller/types/entry"
+	"github.com/momentum-xyz/ubercontroller/types/generic"
+	"github.com/momentum-xyz/ubercontroller/universe"
+	"github.com/momentum-xyz/ubercontroller/utils"
+)
+
+type Calendar struct {
+	ctx      context.Context
+	world    universe.World
+	timerSet *generic.TimerSet[string]
+}
+
+type Event struct {
+	SpaceID *uuid.UUID
+	Title   string
+	Start   time.Time
+	End     time.Time
+	EventID string
+}
+
+var pluginID = uuid.MustParse("f0f0f0f0-0f0f-4ff0-af0f-f0f0f0f0f0f0")
+var attributeName = "events"
+var log = logger.L()
+
+func NewCalendar() *Calendar {
+	calendar := &Calendar{
+		timerSet: generic.NewTimerSet[string](),
+	}
+
+	return calendar
+}
+
+func (c *Calendar) Initialize(ctx context.Context, w universe.World) error {
+	c.ctx = ctx
+	c.world = w
+	return nil
+}
+
+func (c *Calendar) Run() error {
+	fmt.Println("RUN calendar" + c.world.GetID().String())
+
+	c.update()
+
+	return nil
+}
+
+func (c *Calendar) update() {
+	spaces := c.world.GetSpaces(true)
+	events := getAllEvents(spaces)
+	nextEvents := findNextEvents(events)
+
+	c.timerSet.StopAll()
+
+	for _, e := range nextEvents {
```

Comment:
nit: use index and `nextEvents[i]` instead of `_, e`

Reply:
Fixed